Create multiple-config-instances.md (#900)
ZupIT/beagle-docs: content/pt/android/tutorials/multiple-config-instances.md

---
title: Como usar várias instâncias do beagle
weight: 234
description: >-
  Nesta seção, você encontrará o passo a passo para iniciar a configuração do beagle para várias instâncias
---

---
### O que é isso?

> O **Beagle Multiple Instances** é uma maneira de iniciar o beagle para o fluxo atual orientado pelo servidor usando outra configuração do beagle, como novos widgets, ações e assim por diante.

Reviewer: ```suggestion
**Beagle Multiple Instances** é um recurso que permite a inicialização de fluxos server driven distintos com configurações de Beagle independentes. Isso permite que cada fluxo tenha seus próprios componentes, base url, ações, http client, etc.
```

> Você verá abaixo como configurar as várias instâncias.

Reviewer: ```suggestion

```

        it.logInfo("$CONFIG_NAME:analyticsProvider")
        AppAnalytics()
    }

    override val urlBuilder = beagleConfigFactory<UrlBuilder> {
        it.logInfo("$CONFIG_NAME:urlBuilder")
        AppUrlBuilder()
    }

    override fun registeredWidgets() =
        beagleConfigFactory {
            it.logInfo("$CONFIG_NAME:registeredWidgets")
            listOf(
                Text3::class.java as Class<WidgetView>,
                Input::class.java as Class<WidgetView>)
        }

    override fun registeredActions() =
        beagleConfigFactory {
            it.logInfo("$CONFIG_NAME:registeredActions")
            listOf(
                br.com.zup.beagle.sample.actions.CustomAndroidAction::class.java as Class<Action>,
            )
        }

    override fun registeredOperations() =
        beagleConfigFactory {
            it.logInfo("$CONFIG_NAME:registeredOperations")
            mapOf<String, Operation>(
                "isValidCpf" to br.com.zup.beagle.sample.operations.IsValidCPFOperation(),

                )
        }
}
```

### Configurando um ponto de entrada beagle para usar a nova configuração

> Uma função chamada [**newServerDrivenIntent**]({{< ref path="/android/tutorials/how-to-display-a-screen" lang="en" >}}) é responsável por exibir uma tela, agora você pode definir que esta tela usará a configuração definida acima assim:

Reviewer: ```suggestion
A função [**newServerDrivenIntent**]({{< ref path="/android/tutorials/how-to-display-a-screen" lang="en" >}}) é responsável por exibir uma tela server driven. Para definir que esta tela usará a configuração definida acima, faça:
```
